### PR TITLE
Refactor FXIOS-5101 [v115] update theming in url bar view

### PR DESCRIPTION
**Pull request by petruSt24**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5101)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/12164)

Removed references to Color.Photon and made the classes comply to the new theming protocol

### Pull requests checks where applicable
- [X] Fill in the three TODOs above (tickets number and description of your work)
- [X] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by lmarceau** [comment]
@petruSt24 can you put screenshot of the changes so designer can look into the changes to see if we comply?

**Comment by lmarceau** [comment]
Tagging @cwzilla for design review

**Comment by OrlaM** [comment]
Should also include screenshots of private browsing and iPad as there are differences in both use cases. The url bar has a lot going on 😓

**Comment by petruSt24** [comment]
> Should also include screenshots of private browsing and iPad as there are differences in both use cases. The url bar has a lot going on 😓

Yes this is a much bigger component than I expected but it's fun to see how everything connects and falls in place. I actually stopped midway with the screenshots because I noticed the Search Engine image does not comply with the new design for dark mode so I started looking into it to fix, but the image is loaded from a xml. I'm not sure if I should add the dark mode image in the xml or to refactor the search engines to load the image from assets and make them comply with getThemedImageName(name:). Also I remembered about another [issue](https://mozilla-hub.atlassian.net/browse/FXIOS-5498) that I discussed with QA in relation to another ticket and I saw the comment from Nish so I'll be checking this one as well. While looking into getThemedImageName(name:) I wonder why this approach was chosen for themed images instead of just adding both Light/Dark variants to the same asset and with the same name like in [here](https://developer.apple.com/documentation/uikit/uiimage/providing_images_for_different_appearances) since even if the plan is to add multiple different themes in the future the images will probably have 2 variants for light/dark that can get a different tint based on the chosen theme, unless the composing colors will wildly differ between themes.

**Comment by OrlaM** [comment]
Yeah this is probably the most complex view in the entire app 😅 
Laurie put this doc together for a project that was happening last year which documents all of the the different use cases and states of the url bar, it might be helpful https://docs.google.com/document/d/1yni_wctJHYQCYmT3tVt6wbtI73X6kbDATqeq-qDfVrA

For the image themeing, at the time a project called colorways was supposed be starting soon where the app would support many different themes so instead of leaning into some of the tools apple provides to make it convenient to support just light/dark we handled everything in a way that new themes could easily be added later. That project was put on hold so didn't happen in the end, but it might come back around in the future.

**Comment by petruSt24** [comment]
Thank you for the doc, I definitely missed a few use cases when testing. About the Search Engines images, what path should I take in handling them? The Apple way would work for any number of themes for icon style images (like bookmarks, downloads, share) just by applying a tint, but would add difficulties in more artsy style images like the logo or lock_blocked so I guess a universal approach beats having to handle images in 2 ways to avoid edge cases.

**Comment by lmarceau** [comment]
I created another task for the search engine icons theming: https://mozilla-hub.atlassian.net/browse/FXIOS-6318

File: Client/Frontend/Browser/TopTabsViewController.swift
Comment (petruSt24) [reply Client/Frontend/Browser/TopTabsViewController.swift:14]: this change is from another PR, it's already merged to main and shouldn't have appeared here

**Comment by petruSt24** [comment]
> I created another task for the search engine icons theming: https://mozilla-hub.atlassian.net/browse/FXIOS-6318

I'll change just the google image to one without a background to see how it looks and will take care of all the other search engines as part of this task

**Comment by petruSt24** [comment]
Screen Shot 2023-05-05 at 2 39 42 PM
Screen Shot 2023-05-05 at 2 39 44 PM
Screen Shot 2023-05-05 at 2 39 46 PM
Screen Shot 2023-05-05 at 2 39 52 PM
Screen Shot 2023-05-05 at 2 39 54 PM
Screen Shot 2023-05-05 at 2 40 03 PM
Screen Shot 2023-05-05 at 2 40 08 PM
Screen Shot 2023-05-05 at 2 40 18 PM
Screen Shot 2023-05-05 at 2 40 28 PM
Screen Shot 2023-05-05 at 2 40 37 PM
Screen Shot 2023-05-05 at 2 40 44 PM
Light Mode iPhone

**Comment by petruSt24** [comment]
Screen Shot 2023-05-05 at 2 04 28 PM
Screen Shot 2023-05-05 at 2 04 31 PM
Screen Shot 2023-05-05 at 2 04 33 PM
Screen Shot 2023-05-05 at 2 04 56 PM
Screen Shot 2023-05-05 at 2 05 04 PM
Screen Shot 2023-05-05 at 2 05 28 PM
Screen Shot 2023-05-05 at 2 05 31 PM
Screen Shot 2023-05-05 at 2 38 31 PM
Screen Shot 2023-05-05 at 2 38 37 PM
Light Mode iPhone Private Mode

**Comment by petruSt24** [comment]
Screen Shot 2023-05-05 at 1 58 48 PM
Screen Shot 2023-05-05 at 1 58 59 PM
Screen Shot 2023-05-05 at 1 59 01 PM
Screen Shot 2023-05-05 at 1 59 12 PM
Screen Shot 2023-05-05 at 1 59 33 PM
Screen Shot 2023-05-05 at 1 59 52 PM
Screen Shot 2023-05-05 at 2 00 11 PM
Screen Shot 2023-05-05 at 2 00 15 PM
Screen Shot 2023-05-05 at 2 00 19 PM
Screen Shot 2023-05-05 at 2 00 22 PM
Screen Shot 2023-05-05 at 2 00 25 PM
Screen Shot 2023-05-05 at 2 00 39 PM
Screen Shot 2023-05-05 at 2 01 15 PM
 
Light Mode iPad

**Comment by petruSt24** [comment]
Screen Shot 2023-05-05 at 1 46 34 PM
Screen Shot 2023-05-05 at 1 47 18 PM
Screen Shot 2023-05-05 at 1 47 43 PM
Screen Shot 2023-05-05 at 1 48 02 PM
Screen Shot 2023-05-05 at 1 48 22 PM
Screen Shot 2023-05-05 at 1 54 44 PM
Screen Shot 2023-05-05 at 1 54 52 PM
Screen Shot 2023-05-05 at 1 55 00 PM
Screen Shot 2023-05-05 at 1 55 09 PM
Screen Shot 2023-05-05 at 1 55 14 PM
Screen Shot 2023-05-05 at 1 55 35 PM
Screen Shot 2023-05-05 at 1 55 39 PM
Screen Shot 2023-05-05 at 1 55 46 PM
Dark Mode iPad

**Comment by petruSt24** [comment]
Screen Shot 2023-05-05 at 1 36 13 PM
Screen Shot 2023-05-05 at 1 36 23 PM
Screen Shot 2023-05-05 at 1 36 33 PM
Screen Shot 2023-05-05 at 1 36 40 PM
Screen Shot 2023-05-05 at 1 36 58 PM
Screen Shot 2023-05-05 at 1 37 02 PM
Screen Shot 2023-05-05 at 1 37 05 PM
Screen Shot 2023-05-05 at 1 37 59 PM
Screen Shot 2023-05-05 at 1 38 08 PM
Screen Shot 2023-05-05 at 1 38 20 PM
Dark Mode Private Browsing

**Comment by petruSt24** [comment]
Screen Shot 2023-05-05 at 1 25 29 PM
Screen Shot 2023-05-05 at 1 25 48 PM
Screen Shot 2023-05-05 at 1 25 58 PM
Screen Shot 2023-05-05 at 1 26 13 PM
Screen Shot 2023-05-05 at 1 26 25 PM
Screen Shot 2023-05-05 at 1 26 31 PM
Screen Shot 2023-05-05 at 1 26 36 PM
Screen Shot 2023-05-05 at 1 27 42 PM
Screen Shot 2023-05-05 at 1 28 14 PM
Screen Shot 2023-05-05 at 1 30 50 PM
Dark Mode iPhone

File: Client/Assets/Search/SearchPlugins/google-b-1-m.xml
Comment (lmarceau) [inline Client/Assets/Search/SearchPlugins/google-b-1-m.xml:8]: Shouldn't search engine icon changes be in another PR?

File: Client/Frontend/Browser/TopTabsViewController.swift
Comment (lmarceau) [reply Client/Frontend/Browser/TopTabsViewController.swift:14]: I think this is removing that fix, can you make sure we don't remove the `UIDevice.current.userInterfaceIdiom == .pad ? 130` ?

File: Client/Frontend/Widgets/AutocompleteTextField.swift
Comment (lmarceau) [inline Client/Frontend/Widgets/AutocompleteTextField.swift:348]:
```suggestion
        if autocompleteTextLabel?.attributedText != nil {
            autocompleteTextLabel?.backgroundColor = backgroundColor
           autocompleteTextLabel?.textColor = textColor
```

File: Client/Assets/Search/SearchPlugins/google-b-1-m.xml
Comment (petruSt24) [reply Client/Assets/Search/SearchPlugins/google-b-1-m.xml:8]: this doesn't change how the Search Engines work, like having theme-able images, it just replaces the Google logo image with one without a background so it conforms with the design

File: Client/Frontend/Widgets/AutocompleteTextField.swift
Comment (petruSt24) [reply Client/Frontend/Widgets/AutocompleteTextField.swift:348]: I was working on a mac mini for a while and having self. was the only way to assess if a variable is local/global at a glance, until Xcode would load the colors and show everything. Could still happen on slower/older computers or projects that take forever to build.

File: Client/Assets/Search/SearchPlugins/google-b-1-m.xml
Comment (lmarceau) [reply Client/Assets/Search/SearchPlugins/google-b-1-m.xml:8]: Oh ok, that image was provided by design?

Comment (petruSt24) [reply Client/Assets/Search/SearchPlugins/google-b-1-m.xml:8]: indirectly, it was stolen from https://www.figma.com/file/K9m9aQx40RmWnXnIc1GDeQ/iOS-%26-iPad-UI?node-id=1162-108085&t=sT3eJyyZgh4PLTip-0

**Review by lmarceau — APPROVED** [review]
We would just need @cwzilla review, but LGTM

**Comment by cwzilla** [comment]
@petruSt24 I'm going through the design review now, but had some questions before continuing. Some images are hard for me to tell whether the icon is intended to be disabled or tappable.

For example, in this one the Refresh icon appears to be disabled in comparison with the Share icon:
236449605-c38ab375-5b56-4f03-8ce2-1f51aaec322a
However, in this one, the Refresh icon and the Share icon both look disabled compared to the Reader icon, but that would contradict the first example. 
236449617-638ecd63-1aa9-4306-91a9-8d21cb370118

This might be tedious, but is there a way to add in the descriptions of each screenshot to clarify the intended states for each screenshot or if you could guide me to view which parts of the code and what theming tokens we are using for each screenshot would be helpful! Or we can chat on Slack (I couldn't find you). Thanks!

**Comment by cwzilla** [comment]
@petruSt24 and I had a call to review the screenshots and these are the changes we've discussed:
- Update the back/forward disabled color to iconDisabled
- Update the reader mode, share, and refresh icon colors to iconSecondary
- Update the autocomplete text highlight color (in private mode) to layerAccentPrivateNonOpaque

Everything else looks great. Thank you! 😊

**Comment by OrlaM** [comment]
@petruSt24 Looks like the unit test target doesn't compile. Please always verify that unit tests run and pass locally before pushing.

**Comment by lmarceau** [comment]
BR UI tests failed, can you rebase with `main` @petruSt24? I think this will solve the problem

**Comment by petruSt24** [comment]
> BR UI tests failed, can you rebase with `main` @petruSt24? I think this will solve the problem

I'll have another call with Crystal to go over the design one last time and see if no other changes are needed and then I'll rebase to solve this issue

File: Client/Frontend/Toolbar+URLBar/ToolbarButton.swift
Comment (lmarceau) [inline Client/Frontend/Toolbar+URLBar/ToolbarButton.swift:13]: This isn't using the new theming system 🤔 This is a fixed color that we'll deprecate (any colors directly used with `UIColor.`

File: Client/Frontend/Toolbar+URLBar/TabLocationView.swift
Comment (lmarceau) [inline Client/Frontend/Toolbar+URLBar/TabLocationView.swift:29]: Oh, that one too

**Review by lmarceau — APPROVED** [review]

**Comment by lmarceau** [comment]
@petruSt24 can you rebase with latest main fcea6263b6e0def1c836c56d509ddaf1604c35ec ? Still some failing issues to run BR

**Comment by lmarceau** [comment]
Just realized this isn't a flaky tests. It's failing due to the changes in this PR. Can you look into this?
You can fix this issue with the following change: https://github.com/mozilla-mobile/firefox-ios/compare/main...lmarceau:firefox-ios:lm/FXIOS-6469-flaky-test

Screenshot 2023-05-25 at 11 41 12 AM

**Pull request merged**
